examples/seismic/model.py

```diff
@@ -689,29 +763,26 @@ def __init__(self, origin, spacing, shape, space_order, vp, vs, rho, nbpml=20,
         self.damp = Function(name="damp", grid=self.grid)
         initialize_damp(self.damp, self.nbpml, self.spacing, mask=True)
 
-        # Create square slowness of the wave as symbol `m`
-        if isinstance(vp, np.ndarray):
-            self.vp = Function(name="vp", grid=self.grid, space_order=space_order)
-            initialize_function(self.vp, vp, self.nbpml)
-        else:
-            self.vp = Constant(name="vp", value=vp)
-        self._physical_parameters = ('vp',)
+        physical_parameters = []
 
-        # Create square slowness of the wave as symbol `m`
-        if isinstance(vs, np.ndarray):
-            self.vs = Function(name="vs", grid=self.grid, space_order=space_order)
-            initialize_function(self.vs, vs, self.nbpml)
-        else:
-            self.vs = Constant(name="vs", value=vs)
-        self._physical_parameters += ('vs',)
+        self.vp = self._gen_phys_param(vp, 'vp', space_order)
+        physical_parameters.append('vp')
 
-        # Create square slowness of the wave as symbol `m`
-        if isinstance(rho, np.ndarray):
-            self.rho = Function(name="rho", grid=self.grid, space_order=space_order)
-            initialize_function(self.rho, rho, self.nbpml)
+        self.vs = self._gen_phys_param(vs, 'vs', space_order)
+        physical_parameters.append('vs')
+
+        self.rho = self._gen_phys_param(rho, 'rho', space_order)
+        physical_parameters.append('rho')
+
+        self._physical_parameters = as_tuple(physical_parameters)
+
+    def _gen_phys_param(self, field, name, space_order):
```

Comment:
@rhodrin @mloubout not necessarily in this PR, but...can we use this function all over the place please?